Mentoring  Program/BussinessLogic/AdoNetHelper.cs

```diff
@@ -0,0 +1,153 @@
+
+        public AdoNetHelper(IConfiguration config)
+        {
+            this.config = config;
+        }
+
+        public List<Models.Game> GetGames()
+        {
+            var games = new List<Models.Game>();
+            var conectionString = config.GetValue<string>("ConnectionStrings:DefaultConnection");
+
+            string sql = "Select Game.ID, Game.Name, Description, SystemRequirenments, Genres, InterfaceLanguages, FullSupportLanguage, ReleaseDate, IsFreeToPLay, Price, GameCreator.Name as 'CreatorName', GamePublisher.Name as 'PublisherName' FROM Game INNER JOIN GameCreator ON(Game.DeveloperID = GameCreator.ID) INNER JOIN GamePublisher ON(Game.PublisherID = GamePublisher.ID)";
+
+            using (var connection = new SqlConnection(conectionString))
+            {
+                var command = new SqlCommand(sql, connection);
+                var dataReader = command.ExecuteReader();
+                if (dataReader.HasRows)
+                {
+                    while (dataReader.Read())
+                    {
+                        games.Add(MapGame(dataReader));
+                    }
+                }
+
+                dataReader.Close();
+            }
+
+            return games;
+        }
+
+        public Models.Game GetGame(int id)
+        {
+            var game = new Models.Game();
+            var conectionString = config.GetValue<string>("ConnectionStrings:DefaultConnection");
+
+            string sql = $"Select Game.ID, Game.Name, Description, SystemRequirenments, Genres, InterfaceLanguages, FullSupportLanguage, ReleaseDate, IsFreeToPLay, Price, GameCreator.Name as 'CreatorName', GamePublisher.Name as 'PublisherName' FROM Game INNER JOIN GameCreator ON(Game.DeveloperID = GameCreator.ID) INNER JOIN GamePublisher ON(Game.PublisherID = GamePublisher.ID) WHERE Game.ID={id}";
+
+            using (var connection = new SqlConnection(conectionString))
+            {
+                var command = new SqlCommand(sql, connection);
+                var dataReader = command.ExecuteReader();
+                if (dataReader.HasRows)
+                {
+                    if (dataReader.Read())
+                    {
+                        game = MapGame(dataReader);
+                    }
+                }
+
+                dataReader.Close();
+            }
+
+            return game;
+        }
+
+        public bool UpdateGame(int id, Models.Game game)
+        {
+            var conectionString = config.GetValue<string>("ConnectionStrings:DefaultConnection");
+
+            string sql = $"UPDATE Game SET Name = {game.Name}, Description = {game.Description}, SystemRequirenments = {game.SystemRequirenments}, Genres = {string.Join(",", game.Genres)}, InterfaceLanguages = {string.Join(",", game.InterfaceLanguages)}, FullSupportLanguage = {string.Join(",", game.FullSupportLanguage)}, ReleaseDate = {game.ReleaseDate}, IsFreeToPLay = {game.IsFreeToPlay}, Price = {game.Price} WHERE ID = {id}";
```

Comment:
Looks like a target for SQL injection. What if someone pass as Name something;DROP TABLE Games?

Reply:
Update scripts by using parameters